base/linalg/uniformscaling.jl

```diff
@@ -167,6 +167,8 @@ end
 
 inv(J::UniformScaling) = UniformScaling(inv(J.λ))
 norm(J::UniformScaling, p::Real=2) = abs(J.λ)
+det(J::UniformScaling{T}) where {T} = J.λ==one(T) ? one(T) : ArgumentError("Determinant of UniformScaling is only well-defined when λ=1")
```

Comment:
`isone(J.λ)  ? one(T) : throw(ArgumentError("Determinant of UniformScaling is only well-defined when λ = 1."))`